### DISCUSSION TITLE
Not getting suggestions using astronvim

### DISCUSSION THREAD

**Discussion by oriongonza** [community/community#63324] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

I'm using copilot, I'm authenticated and all
But I'm getting almost no suggestions, many many fewer than when I'm using vscode

Using lazy

```lua
{
    "github/copilot.vim",
    event = "VeryLazy",
    autoStart = true,
  }
 ```
When I run :Copilot panel it works.

**Comment by Akash1134** [comment]
Hi @dev-ardi ,

First, make sure that you are using the most recent versions of VSCode and the Copilot extension. Next, ensure that you have "editor.inlineSuggest.enabled": true in your Github Copilot extension settings. You can access these with the gear symbol on the extension page in your editor.

If that does not help, please share the below info:

- The output from View > Command Palette... > Github Copilot: Collect Diagnostics
- Any output you see in the bottom output panel with "Github Copilot" selected from the dropdown (while working on some code). 
- Any information about the network connection/environment you are working in, like VPN/Firewall/Proxy/VM.

You may also refer the [troubleshooting doc](https://docs.github.com/en/copilot/configuring-github-copilot/configuring-github-copilot-in-your-environment) to troubleshoot Copilot in your IDE.

Thanks!

**Reply by oriongonza** [reply]
I'm not using vscode, I'm using neovim.

No VPN either

**Comment by shaunemslie** [comment]
Hi @dev-ardi,

After reading through the docs for `lazy.nvim` I observed that the `event` property allows you to lazily load a plugin after an event is triggered, e.g `BufEnter`
I'm curious as to whether the `VeryLazy` event may be causing the issue. I'm assuming this is a custom-defined autocmd, if not perhaps an event from lazy? If I could maybe get more information surrounding that event I may be able help.
Thanks in advance.

**Edit:** After some more research, I've ascertained that the `VeryLazy` event is defined by lazy.nvim and triggers after the `VimEnter` & initial load events. I doubt this would result in fewer suggestions. Falling back on below request for additional information. Perhaps an incorrect LSP configuration? Furthering research on Astronvim.

**Reply by shaunemslie** [reply]
Additionally, have you perhaps noticed any specific patterns as to when suggestions are fewer? Maybe specific filetypes and the likes?

**Reply by oriongonza** [reply]
https://github.com/folke/lazy.nvim#-user-events VeryLazy loads after all else is done. Even if I change it to something BufEnter it has the same behavior.

> Additionally, have you perhaps noticed any specific patterns as to when suggestions are fewer? Maybe specific filetypes and the likes?

Honestly getting a suggestion is so rare that I can't really see a pattern.

**Reply by oriongonza** [reply]
I've tried to stop using astronvim and I still get no suggestions. Panel working fine still. zbirenbaum/copilot.lua doesn't work either.

**Reply by oriongonza** [reply]
Today I got suggestions for like 15 seconds at the rate I expected and it stopped working. This is extremely confusing

**Comment by Franklyn-R-Silva** [comment]
It seems like you're encountering an issue with GitHub Copilot not providing as many suggestions as expected when using it with Vim. Here are a few suggestions to help you troubleshoot and potentially resolve the issue:

1. **Check Configuration:** Make sure that your configuration for Copilot in Vim is correct. The code snippet you provided appears to be configuring Copilot to auto-start in "VeryLazy" mode, which might affect its suggestion behavior. You might want to try other modes or adjust the settings to see if it impacts the suggestions.

2. **Check Authentication:** Double-check that your GitHub Copilot is properly authenticated with your GitHub account. Ensure that your GitHub account is linked correctly in the Copilot extension.

3. **Network Connectivity:** Copilot needs a stable internet connection to work properly since it processes your code with the help of an external server. Make sure your network connection is stable and not blocked by any firewalls or proxies that might impede Copilot's communication.

4. **Plugin Updates:** Ensure that both the GitHub Copilot plugin and the Vim plugin you're using are up-to-date. New updates might contain bug fixes or improvements that could resolve the issue you're facing.

5. **Open an Issue:** If you've tried the above steps and the issue persists, consider opening an issue on the GitHub Copilot repository. The developers and community might be able to provide more specific help or insights into the problem.

6. **VSCode Comparison:** It's normal for GitHub Copilot to have some variations in behavior between different code editors. The level of integration might differ, and certain features might work better in one editor compared to another.

7. **Documentation:** Refer to the official documentation for both GitHub Copilot and the Vim plugin you're using. There might be specific configuration options or troubleshooting steps mentioned there that could help.

8. **Community Forums:** Check if there are any community forums or discussions related to GitHub Copilot and its integration with Vim. Other users might have encountered similar issues and found solutions.

Remember that debugging technical issues can sometimes be a trial-and-error process, and it might require some patience. If all else fails, reaching out to the official support channels of GitHub Copilot or the Vim plugin you're using could provide more personalized assistance based on the specifics of your setup.

**Reply by oriongonza** [reply]
Hello chatGPT! 

While I appreciate the intent to help if you had read the full conversation you would've found that none of this tips apply here.

**Comment by hovissimo** [comment]
Ditto, except I'm using main line Vim 9.0.1826

I'm authenticated, using the panel works, but I'm getting no suggestions as of 1-2 weeks ago. I had plenty before.

Lazy is not related, it's not in my Vim.

When I remove all of my Copilot related maps and settings in my .vimrc Copilot starts suggesting again (after reloading Vim).  It seems there is some kind of error under the hood regarding what's been mapped/configured, and the problem is not being detected by the `:Copilot suggest` tool.

**Comment by hovissimo** [comment]
The line in my config that appears to be breaking Copilot is
```vim
let g:copilot_no_tab_map = v:true
```

Note that this line is suggested in the Copilot.vim help documentation!
```
                                                copilot-i_<Tab>
Copilot.vim uses <Tab> to accept the current suggestion.  If you have an
existing <Tab> map, that will be used as the fallback when no suggestion is
displayed.

                                                copilot#Accept()
If you'd rather use a key that isn't <Tab>, define an <expr> map that calls
copilot#Accept().  Here's an example with CTRL-J:

        imap <silent><script><expr> <C-J> copilot#Accept("\<CR>")
        let g:copilot_no_tab_map = v:true
```

It makes sense that this flag should disable the <tab> mapping, and perhaps also cause `:Copilot status` to complain, but it's surprising and probably buggy that it disables inline suggestions.

#### Apparently my Copilot has been broken since I copied this suggestion from Copilot.vim's own documentation.

**Comment by tpope** [comment, marked as answer]
This should be fixed in Copilot.vim 1.10.2. Sorry about that!

**Reply by hovissimo** [reply]
Tyvm!

**Reply by oriongonza** [reply]
Thanks Tim!

**Comment by eleijonmarck** [comment]
Else i am authenticatied. Any suggestions if this is actually something that i should care about?

I am getting `Copilot: <Tab> map has been disabled or is claimed by another plugin`

from :Copilot status. Is there something wrong with my setup? I have a freshfresh setup with astronvim and the only thing i did was to add the configuration to a file `copilot.lua` in the `~/.config/nvim/lua/plugins/copilot.lua`
